packages/web-components/fast-foundation/src/tree-item/tree-item.ts

```diff
@@ -0,0 +1,259 @@
+import {
+    attr,
+    FASTElement,
+    Notifier,
+    Observable,
+    observable,
+} from "@microsoft/fast-element";
+import {
+    getDisplayedNodes,
+    isHTMLElement,
+    keyCodeArrowDown,
+    keyCodeArrowLeft,
+    keyCodeArrowRight,
+    keyCodeArrowUp,
+    keyCodeEnter,
+    keyCodeSpace,
+} from "@microsoft/fast-web-utilities";
+import { StartEnd } from "../patterns/start-end";
+import { applyMixins } from "../utilities/apply-mixins";
+
+export class TreeItem extends FASTElement {
+    @attr({ mode: "boolean" })
+    public expanded: boolean = false;
+
+    @attr({ mode: "boolean" })
+    public selected: boolean;
+
+    @attr({ mode: "boolean" })
+    public disabled: boolean;
+
+    public expandCollapseButton: HTMLDivElement;
+
+    public treeItem: HTMLElement;
+
+    @observable
+    public focusable: boolean = false;
+
+    @observable
+    public hasItems: boolean;
+
+    public items: HTMLSlotElement;
+    public handleItemsChange(): void {
+        // we only want to project the slot of there will be items
+        this.hasItems = this.querySelectorAll("[slot='item']").length > 0;
+
+        if (this.$fastController.isConnected) {
+            this.items.assignedNodes().forEach((node: HTMLElement) => {
+                if (node instanceof TreeItem) {
+                    (node as TreeItem).nested = true;
+                }
+            });
+        }
+    }
+
+    @observable
+    public nested: boolean;
+
+    @observable
+    public renderCollapsedChildren: boolean = false;
+
+    private notifier: Notifier;
+
+    constructor() {
+        super();
+
+        this.handleItemsChange();
+    }
+
+    private getParentTreeNode() {
+        const parentNode: Element | null | undefined = this.parentElement!.closest(
+            "[role='tree']"
+        );
+        return parentNode;
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+
+        const parentTreeNode: any = this.getParentTreeNode();
+        if (parentTreeNode) {
+            this.notifier = Observable.getNotifier(parentTreeNode);
+            this.notifier.subscribe(this, "render-collapsed-nodes");
```

Comment:
I think you should subscribe to the property name rather than the attribute name.

Reply:
Subscribing to property name now